### DISCUSSION TITLE
Can't retrieve list of members from a given role

### DISCUSSION THREAD

**Discussion by storca** [Rapptz/discord.py#6000] [Q&A] [answered]
Hello

I'm having trouble retrieving a list of members from a given role.
I retrieve the given list by calling myrole.members and most of the time it is empty [] (but there are users that have the role!).
However, when the bot in connected to the voicechat, and when I call the command in the that, it adds himself in the list.

I really don't understand, looks like black magic to me, is it a permission issue ?

Any help/tricks/tips would be greatly appreciated

**Comment by SebbyLaw** [comment, marked as answer]
See #5867.

To further elaborate:

Discord recently released a mandatory update to their gateway requiring the usage of gateway intents. You can find documentation on this [here](https://discordpy.readthedocs.io/en/latest/intents.html).

Member related events will no longer work due to this change by default. Since these are [privileged intents](https://discordpy.readthedocs.io/en/latest/intents.html#privileged-intents) you need to both enable it in your portal and in your code. For more info see the ["Where’d my members go?"](https://discordpy.readthedocs.io/en/latest/intents.html#where-d-my-members-go) section in the documentation.

**Reply by storca** [reply]
That fixed it for me!

I'll try to add this in the documentation in the Role.members section when I have time to do so.

Thank you very much for your help! :D
